File: Week 6/sentimental-readability/readability.py
def main():
    uinput = input("What is your input? ")

    myCount = count(uinput)

    lvalue = (myCount['lettercount'] / myCount['wordcount']) * 100
    svalue = (myCount['sentencecount'] / myCount['wordcount']) * 100

    index = round((0.0588 * lvalue - 0.296 * svalue - 15.8))

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


def count(a):
    counts = {'lettercount': 0, 'wordcount': 1, 'sentencecount': 0}
    for i in a:
        if i.isalpha():
            counts['lettercount'] += 1
        elif i.isspace():
            counts['wordcount'] += 1
        elif i in ['.', '!', '?']:
            counts['sentencecount'] += 1
    return counts


if __name__ == "__main__":
    main()

File: Week 2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

typedef struct
{
    int lettercount;
    int wordcount;
    int sentencecount;
} Count;

Count count(string a);

int main(void)
{
    string input = get_string("What is your input? ");

    Count myCount = count(input);

    float lvalue = ((float) myCount.lettercount / myCount.wordcount) * 100;
    float svalue = ((float) myCount.sentencecount / myCount.wordcount) * 100;

    int index = round((0.0588 * lvalue - 0.296 * svalue - 15.8));

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", index);
    }
}

Count count(string a)
{
    Count counts = {0, 1, 0};
    for (int i = 0; a[i] != '\0'; i++)
    {
        if (isalpha(a[i]))
        {
            counts.lettercount++;
        }
        else if (isspace(a[i]))
        {
            counts.wordcount++;
        }
        else if (a[i] == '.' || a[i] == '!' || a[i] == '?')
        {
            counts.sentencecount++;
        }
    }
    return counts;
}
